lib/rucio/daemons/common.py

```diff
@@ -23,22 +23,133 @@
 import time
 from collections.abc import Callable, Generator, Iterator, Sequence
 from typing import Any, Generic, Optional, TypeVar, Union
+from types import TracebackType, FrameType
+from abc import ABC, abstractmethod
```

Comment:
To avoid doing this kind of work manually (which would be a waste of time), would it make sense to enforce something like `isort` as part of the pre-commit hook? https://pycqa.github.io/isort/

Reply:
That’s an interesting proposal. I would recommend opening a separate issue to discuss this and to consider applying it across the entire codebase beforehand.

Reply:
I had opened this some weeks ago https://github.com/rucio/rucio/issues/6488 where I mentioned `isort` as well - I will bring it up in the retreat and see what the overall thoughts are. For now, I will make this change manually